src/parser/Parser.ts

```diff
@@ -200,6 +200,13 @@ export class Parser {
         let current = 0;
         let tokens = toParse;
 
+        //the depth of the calls to function declarations. Helps some checks know if they are at the root or not.
```

Comment:
I played around with `Library` statements to confirm something I think I heard but can't find an authoritative source for: that `Library` statements must only be at the top of the file, and can't occur after any non-library statements (including named function declarations).  Running a quick test against RBI with this file confirms that:

```brightscript
Library "v30/bslCore.brs"

sub main()
    print "Hello, world!"
end sub

Library "v30/bslCore.brs"  ' <-- syntax error on line 7
```

Looks like we'll need to replace the depth tracking logic with a simple "have we done anything that's not a library yet?" check :cry:  I'm sorry it took so long to find this!  Somewhere within the `declaration()` function feels reasonable for this, but I haven't tested that at all.

Reply:
I already handle this scenario [here](https://github.com/sjbarag/brs/pull/216/files#diff-0af3c0ebba8c6d54ca06e99fa694be4fR666) 

```
//libraries must be at the very top of the file before any other declarations.
let isAtTopOfFile = true;
for (let statement of statements) {
    //if we found a non-library statement, this statement is not at the top of the file
    if (!(statement instanceof Stmt.Library)) {
        isAtTopOfFile = false;
    }
}
```

Reply:
Derp, I missed that!  Thanks for pointing it out :smiley: